instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/internal/AwsXrayEnvSpanLinksExtractor.java

```diff
@@ -42,22 +43,39 @@ public void extract(
   }
 
   public static void extract(SpanLinksBuilder spanLinks) {
-    String parentTraceHeader = System.getenv(AWS_TRACE_HEADER_ENV_KEY);
-    if (parentTraceHeader == null || parentTraceHeader.isEmpty()) {
+    Map<String, String> contextMap = getSystemPropertyOrEnvironmentVariable();
+    if (contextMap.isEmpty()) {
       return;
     }
     Context xrayContext =
-        AwsXrayPropagator.getInstance()
-            .extract(
-                Context.root(),
-                Collections.singletonMap(AWS_TRACE_HEADER_PROPAGATOR_KEY, parentTraceHeader),
-                MapGetter.INSTANCE);
+        AwsXrayPropagator.getInstance().extract(Context.root(), contextMap, MapGetter.INSTANCE);
     SpanContext envVarSpanCtx = Span.fromContext(xrayContext).getSpanContext();
     if (envVarSpanCtx.isValid()) {
       spanLinks.addLink(envVarSpanCtx, LINK_ATTRIBUTES);
     }
   }
 
+  public static Map<String, String> getSystemPropertyOrEnvironmentVariable() {
```

Comment:
I'd make this method shorter, perhaps something like
```java
String traceHeader = System.getProperty(AWS_TRACE_HEADER_PROP);
if (isEmptyOrNull(traceHeader)) {
  traceHeader = System.getenv(AWS_TRACE_HEADER_ENV_KEY);
}
return isEmptyOrNull(traceHeader) ? Collections.emptyMap() : Collections.singletonMap(AWS_TRACE_HEADER_PROPAGATOR_KEY, traceHeader);
```

Reply:
Excellent suggestion - I've changed this - thanks